app/controllers/multifactor_auths_controller.rb

```diff
@@ -8,15 +8,23 @@ class MultifactorAuthsController < ApplicationController
   def new
     @seed = ROTP::Base32.random_base32
     session[:mfa_seed] = @seed
+    session[:mfa_seed_expire] = Gemcutter::MFA_KEY_EXPIRY.from_now.utc.to_i
     text = ROTP::TOTP.new(@seed, issuer: issuer).provisioning_uri(current_user.email)
     @qrcode_svg = RQRCode::QRCode.new(text, level: :l).as_svg
   end
 
   def create
```

Comment:
This method is too long. I think we can move logic of checking expiry, verifying otp and setting `last_otp_at` to `User#enable_mfa!`[1]. You can use `errors[:base]=` in model and `current_user.errors.full_messages` in controller to pass error messages.

[1] It doesn't mean now `enable_mfa!` has to extra long. what you are doing in `elsif ...` is very similar to `User#verify_digit_otp`, you can use it in `User#enable_mfa!` after some refactoring,